In [11]:
import pandas as pd
import numpy as np

df = pd.read_csv("work/datasets/hotel_bookings.csv")

df.head() # mostra toda a estrutura da tabela
df.shape #mostra o numero de linhas e colunas
df.info() #mostra o tipo do atributo de cada coluna


<class 'pandas.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 32 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  str    
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  str    
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 8   stays_in_week_nights            119390 non-null  int64  
 9   adults                          119390 non-null  int64  
 10  children                        119386 non-null  float64
 11  babies                          119390 non-null  int64  
 12  meal                       

#### 1. Quantas instâncias e quantos atributos existem?
Existem 119390 instancias(linhas) e 32 atributos(colunas)

#### 2. Quais atributos são numéricos e quais são categóricos?

In [12]:
display(f"Numericos: {df.select_dtypes(include=["number"]).columns.tolist()}")
display(f"Categoricos: {df.select_dtypes(include=["category", "string", "object"]).columns.tolist()}")

"Numericos: ['is_canceled', 'lead_time', 'arrival_date_year', 'arrival_date_week_number', 'arrival_date_day_of_month', 'stays_in_weekend_nights', 'stays_in_week_nights', 'adults', 'children', 'babies', 'is_repeated_guest', 'previous_cancellations', 'previous_bookings_not_canceled', 'booking_changes', 'agent', 'company', 'days_in_waiting_list', 'adr', 'required_car_parking_spaces', 'total_of_special_requests']"

"Categoricos: ['hotel', 'arrival_date_month', 'meal', 'country', 'market_segment', 'distribution_channel', 'reserved_room_type', 'assigned_room_type', 'deposit_type', 'customer_type', 'reservation_status', 'reservation_status_date']"

#### 3. Quais atributos são binários?

In [13]:
print(f"As colunas binarias sao: {[col for col in df.columns if df[col].nunique() == 2]}")

As colunas binarias sao: ['hotel', 'is_canceled', 'is_repeated_guest']


#### 4. Quais atributos representam características da reserva, do cliente, da  estadia e do hotel?

In [ ]:
print(df.columns.tolist()) #vou pular essa questao devido ao trabalho para categorizar os dados, isso nao tem uma forma de fazer automatica

['hotel', 'is_canceled', 'lead_time', 'arrival_date_year', 'arrival_date_month', 'arrival_date_week_number', 'arrival_date_day_of_month', 'stays_in_weekend_nights', 'stays_in_week_nights', 'adults', 'children', 'babies', 'meal', 'country', 'market_segment', 'distribution_channel', 'is_repeated_guest', 'previous_cancellations', 'previous_bookings_not_canceled', 'reserved_room_type', 'assigned_room_type', 'booking_changes', 'deposit_type', 'agent', 'company', 'days_in_waiting_list', 'customer_type', 'adr', 'required_car_parking_spaces', 'total_of_special_requests', 'reservation_status', 'reservation_status_date']
